File: app/src/main/java/com/beiming/uhf_test/TestActivity.java
package com.beiming.uhf_test;

import android.app.ProgressDialog;
import android.os.AsyncTask;
import android.os.Bundle;
import android.text.TextUtils;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.beiming.uhf_test.bean.LibAssetBean;
import com.beiming.uhf_test.databinding.ActivityTestBinding;
import com.beiming.uhf_test.db.GreenDaoManager;
import com.beiming.uhf_test.fragment.FileImport;
import com.beiming.uhf_test.fragment.FileXls;
import com.beiming.uhf_test.greendao.gen.LibAssetBeanDao;
import com.beiming.uhf_test.tools.UIHelper;
import com.beiming.uhf_test.utils.LogPrintUtil;

import java.util.List;

public class TestActivity extends AppCompatActivity {

    private ActivityTestBinding binding;
    private String s;
    private List<LibAssetBean> list;
    private LibAssetBeanDao dao;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityTestBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        dao = GreenDaoManager.getInstance().getSession().getLibAssetBeanDao();

        binding.b1111.setOnClickListener(view -> {

            new exportExcelTask().execute();

        });

        binding.b2.setOnClickListener(v -> {
            List<LibAssetBean> data = dao.loadAll();
            LogPrintUtil.zhangshi("s:size=" + data.size());
        });
    }

    public class exportExcelTask extends AsyncTask<String, Integer, Boolean> {
        ProgressDialog mypDialog;

        @Override
        protected Boolean doInBackground(String... params) {
            // TODO Auto-generated method stub
            list = FileXls.readXLSToLibAssetList("/sdcard/UHF/bb.xls", 0);
//            dao.insertInTx(list);
            return (null != list);
        }

        @Override
        protected void onPostExecute(Boolean result) {
            super.onPostExecute(result);

            mypDialog.cancel();

            if (result) {
                for (LibAssetBean item : list) {
                    LogPrintUtil.zhangshi("s:" + item.getAssetId());
                }
            } else {

                LogPrintUtil.zhangshi("s:" + "fail");
            }
        }

        @Override
        protected void onPreExecute() {
            // TODO Auto-generated method stub
            super.onPreExecute();

            mypDialog = new ProgressDialog(TestActivity.this);
            mypDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
            mypDialog.setMessage("导出中...");
            mypDialog.setCanceledOnTouchOutside(false);
            mypDialog.show();
        }
    }
}

File: app/src/main/java/com/beiming/uhf_test/TestActivity.kt
package com.beiming.uhf_test

import android.app.DatePickerDialog
import android.os.Build
import android.os.Bundle
import android.widget.DatePicker
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import com.beiming.uhf_test.databinding.ActivityTestBinding
import com.beiming.uhf_test.utils.LogPrintUtil
import com.beiming.uhf_test.utils.TimeUtils
import java.util.*


class TestActivity : AppCompatActivity() {

    lateinit var binding: ActivityTestBinding;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityTestBinding.inflate(layoutInflater)
        setContentView(binding.root)




        binding.b1111.setOnClickListener {
            view ->
            var calendar = Calendar.getInstance()
            var dialog: DatePickerDialog = DatePickerDialog(this, object : DatePickerDialog.OnDateSetListener{
                override fun onDateSet(view: DatePicker?, year: Int, month: Int, dayOfMonth: Int) {
                    val desc: String = java.lang.String.format(
                        "您选择的日期是：%s年%s月%s日",
                        year,
                        month + 1,
                        dayOfMonth
                    )

                    val s = TimeUtils.formatDate(year, month + 1, dayOfMonth)
                    val ts = TimeUtils.toTs(s)

                    LogPrintUtil.zhangshi("ts:" + ts)

                    Toast.makeText(this@TestActivity, s, Toast.LENGTH_SHORT).show()
                }

            },
            calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH))
            dialog.show()

            LogPrintUtil.zhangshi("1111")

        }
        binding.b2.setOnClickListener {
            LogPrintUtil.zhangshi("2222")
        }
    }


}